load_balancer.go

```diff
@@ -118,7 +118,34 @@ func (odbi *ovnDBImp) lbDelImp(name string) (*OvnCommand, error) {
 		Table: tableLoadBalancer,
 		Where: []interface{}{condition},
 	}
-	operations := []libovsdb.Operation{deleteOp}
+
+	// Also delete references from Logical switches
+	row := make(OVNRow)
+	row["name"] = name
+	lbuuid := odbi.getRowUUID(tableLoadBalancer, row)
+	if len(lbuuid) == 0 {
+		return nil, ErrorNotFound
+	}
+	mutateUUID := []libovsdb.UUID{{lbuuid}}
+	mutateSet, err := libovsdb.NewOvsSet(mutateUUID)
+	if err != nil {
+		return nil, err
+	}
+	mutation := libovsdb.NewMutation("load_balancer", opDelete, mutateSet)
+	ucondition, err := odbi.getRowUUIDContainsUUID(tableLogicalSwitch, "load_balancer", lbuuid)
+	if err != nil {
+		return nil, err
+	}
+	mucondition := libovsdb.NewCondition("load_balancer", "==", libovsdb.UUID{ucondition})
```

Comment:
we can delete like that?  in this case, should update loadbalance array in logical switch.

Reply:
this will delete all LB in LP list ? are you sure?

Reply:
could you write a test case for that, I can review.

Reply:
@noah8713 local test i ran before create pr. Main problem that if test fails, cleanup phase not running, so if delete dhcp options for example not running, second iteration of test run always says that object exists.
As for load balancer may be we need you fix for all versions of ovn, because sometimes local test run fails for me with the same error, that load balancer not deleted.

Reply:
@softwarejl : It will only delete LB by matching uuid https://github.com/eBay/go-ovn/pull/33/files#diff-9a905803b6ff1b7d0e209a45f0ae354fR139 and not all